magenta-lib/src/main/scala/magenta/tasks/ASGTasks.scala

```diff
@@ -112,8 +112,16 @@ case class CullInstancesWithTerminationTag(pkg: DeploymentPackage, stage: Stage,
     implicit val elbClient = ELB.client(keyRing, region)
     val instancesToKill = asg.getInstances.filter(instance => EC2.hasTag(instance, "Magenta", "Terminate", ec2Client))
     val orderedInstancesToKill = instancesToKill.transposeBy(_.getAvailabilityZone)
-    reporter.verbose(s"Culling instances: ${orderedInstancesToKill.map(_.getInstanceId).mkString(", ")}")
-    orderedInstancesToKill.foreach(instance => ASG.cull(asg, instance, asgClient, elbClient))
+    try {
+      reporter.verbose(s"Culling instances: ${orderedInstancesToKill.map(_.getInstanceId).mkString(", ")}")
+      orderedInstancesToKill.foreach(instance => ASG.cull(asg, instance, asgClient, elbClient))
+    } catch {
+      case e: AmazonServiceException if desiredSizeReset(e) => {
+        reporter.info(s"${e.getMessage}. Your ASG desired size may have been reset. Please check that appropriate dependencies are included in riff-raff.yaml, or ensure desiredSize isn't set in the Cloudformation.")
```

Comment:
The message could more explicitly talk about the potential root cause being that two parts of the deploy are both trying to modify a cloudformation stack simultaneously. At the moment a solution is suggested without the context of what the problem is.

Also: note that `e` is being swallowed here so the deploy will no longer fail; I think it makes sense to either:
 - report a message (maybe warn instead of info) and then re-throw the exception
 - throw a new exception with this message (nesting the original exception for completeness)